File: Clinicas/Clinicas.Web/bin/Release/PublishOutput/app/services/financeiro.service.js
(function () {
    'use strict';

    angular
           .module('app') // Define a qual módulo seu .service pertence
           .factory('financeiroservice', financeiroservice); //Define o nome a função do seu .service

    financeiroservice.$inject = ['$http', 'servicebase']; //Lista de dependências
    function financeiroservice($http, servicebase) {

        var service = this;

        var service = {
            getContasApagarReceber: getContasApagarReceber,
            getParcelaById: getParcelaById,
            salvarFinanceiro: salvarFinanceiro,
            listarmeiospagamento: listarmeiospagamento,

            getTransferencias: getTransferencias,
            getTransferenciaById: getTransferenciaById,
            saveTransferencia: saveTransferencia,
            excluirTransferencia: excluirTransferencia,
            alterarParcela: alterarParcela,
            excluirParcela: excluirParcela,
            pesquisar: pesquisar,
            gerarparcelas: gerarparcelas,
            
            listarcheques: listarcheques,
            pesquisarCheques: pesquisarCheques,
            excluircheque: excluircheque,
            getChequeById: getChequeById,
            salvarCheque: salvarCheque,
            listarbancos: listarbancos,
            excluirbanco: excluirbanco,
            getBancoById: getBancoById,
            salvarBanco: salvarBanco,

            // contas
            listarContas: listarContas,
            excluirConta: excluirConta,
            getContaById: getContaById,
            saveConta: saveConta,
            pesquisarContas: pesquisarContas,

            // plano de contas 
            listarPlanodeContas: listarPlanodeContas,
            excluirPlanoConta: excluirPlanoConta,
            savePlanoConta: savePlanoConta,
            pesquisarPlanoContas: pesquisarPlanoContas,
            getPlanoContaById: getPlanoContaById,

            getFinanceiroPorId:getFinanceiroPorId

        }
        return service;
        

        function getFinanceiroPorId(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getFinanceiroPorId?id=" + id);
        }

        //Implementação das funções
        //***********************************   plano de conta     ***************************************************

        function listarPlanodeContas() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarPlanodeContas");
        }

        function excluirPlanoConta(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluirPlanoConta?id=" + id);
        }

        function savePlanoConta(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/savePlanoConta", model);
        }

        function pesquisarPlanoContas(nome,codigo,tipo) {
            return $http.get(servicebase.urlApi() + "/financeiro/pesquisarPlanoContas?nome=" + nome+"&codigo="+codigo+"&tipo="+tipo);
        }

        function getPlanoContaById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getPlanoContaById?id=" + id);
        }

        //Implementação das funções
        //***********************************   meio de pagamento     ***************************************************
        function listarmeiospagamento() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarmeiospagamento");
        }

        //Implementação das funções
        //***********************************   conta     ***************************************************
        function listarcontas() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarContas");
        }

        function excluirConta(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluirConta?id=" + id);
        }

        function getContaById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getContaById?id=" + id);
        }

        function saveConta(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/saveConta", model);
        }

        function pesquisarContas(nome, codigo) {
            return $http.get(servicebase.urlApi() + "/financeiro/pesquisarContas?nome=" + nome + "&codigo=" + codigo);
        }


        //***********************************   financeiro     ***************************************************

        function getContasApagarReceber(tipo) {
            return $http.get(servicebase.urlApi() + "/financeiro/getContasApagarReceber?tipo=" + tipo);
        }

        

        function excluirParcela(idFinanceiro, idParcela) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluirParcela?idFinanceiro=" + idFinanceiro + "&idParcela=" + idParcela);
        }

        function pesquisar(tipo, descricao, tipoConta) {
            return $http.get(servicebase.urlApi() + "/financeiro/pesquisar?tipo=" + tipo + "&descricao=" + descricao + "&tipoConta=" + tipoConta);
        }
        

        function listarContas() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarContas");
        }

        function getTransferencias() {
            return $http.get(servicebase.urlApi() + "/financeiro/getTransferencias");
        }

        function getTransferenciaById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getTransferenciaById?id=" + id);
        }

        function excluirTransferencia(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluirTransferencia?id=" + id);
        }

        function getParcelaById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getById?id=" + id);
        }

        function gerarparcelas(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/gerarparcelas", model);
        }

        function saveTransferencia(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/saveTransferencia", model);
        }

        function alterarParcela(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/alterarParcela", model);
        }

        function salvarFinanceiro(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/salvarFinanceiro", model);
        }

        //***********************************   cheques     ***************************************************
        function listarcheques() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarcheques");
        }

        function excluircheque(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluircheque?id=" + id);
        }

        function getChequeById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getChequeById?id=" + id);
        }

        function pesquisarCheques(emitente, banco) {
            return $http.get(servicebase.urlApi() + "/financeiro/pesquisarCheques?emitente=" + emitente + "&banco=" + banco);
        }

        function salvarCheque(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/salvarCheque", model);
        }

        //***********************************   bancos     ***************************************************
        function listarbancos() {
            return $http.get(servicebase.urlApi() + "/financeiro/listarbancos");
        }

        function excluirbanco(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/excluirbanco?id=" + id);
        }

        function getBancoById(id) {
            return $http.get(servicebase.urlApi() + "/financeiro/getBancoById?id=" + id);
        }

        function salvarBanco(model) {
            return $http.post(servicebase.urlApi() + "/financeiro/salvarBanco", model);
        }


        
    }
})();

File: Clinicas/Clinicas.Application/Services/FinanceiroService.cs
using Clinicas.Application.Services.Interfaces;
using Clinicas.Domain.Model;
using Clinicas.Infrastructure.Repository.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Clinicas.Application.Services
{
    public class FinanceiroService : IFinanceiroService
    {
        private readonly IFinanceiroRepository _financeiroRepository;

        public FinanceiroService(IFinanceiroRepository rp)
        {
            _financeiroRepository = rp;
        }

        #region Conta 
        public void SalvarConta(Conta conta)
        {
            _financeiroRepository.SalvarConta(conta);
        }

        public void ExcluirConta(Conta conta)
        {
            _financeiroRepository.ExcluirConta(conta);
        }

        public ICollection<Conta> ListarContas(int idclinica)
        {
            return _financeiroRepository.ListarContas(idclinica);
        }

        public ICollection<Conta> PesquisarContas(int idclinica)
        {
            return _financeiroRepository.ListarContas(idclinica);
        }

        public Conta ObterContaPorId(int id)
        {
            return _financeiroRepository.ObterContaPorId(id);
        }

        public ICollection<Conta> PesquisarContas(string nome, int? codigo,int idclinica)
        {
            return _financeiroRepository.PesquisarContas(nome,codigo, idclinica);
        }
        #endregion

        #region Plano Conta]
        public PlanoConta ObterPlanodeContasPorId(int id)
        {
            return _financeiroRepository.ObterPlanodeContasPorId(id);
        }

        public ICollection<PlanoConta> ListarPlanoContas(int idclinica)
        {
            return _financeiroRepository.ListarPlanoContas(idclinica);
        }

        public ICollection<PlanoConta> PesquisarPlanoContas(string nome, int? codigo, string tipo,int idclinica)
        {
            return _financeiroRepository.PesquisarPlanoContas(nome,codigo,tipo,idclinica);
        }

        public void SalvarPlanoConta(PlanoConta planoConta)
        {
            _financeiroRepository.SalvarPlanoConta(planoConta);
        }

        public void ExcluirPlanoConta(PlanoConta planoConta)
        {
            _financeiroRepository.ExcluirPlanoConta(planoConta);
        }
        #endregion
        
        public void SalvarFinanceiro(Financeiro financeiro)
        {
            _financeiroRepository.SalvarFinanceiro(financeiro);
        }

        public void ExcluirFinanceiro(Financeiro financeiro)
        {
            _financeiroRepository.ExcluirFinanceiro(financeiro);
        }

        public Financeiro ObterFinanceiroPorId(int id)
        {
            return _financeiroRepository.ObterFinanceiroPorId(id);
        }

        public MeioPagamento GetMeioPagamentoPorId(int id)
        {
            return _financeiroRepository.GetMeioPagamentoPorId(id);
        }

        public ICollection<FinanceiroParcela> ListarContasaReceber(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaReceber(idclinica, idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaReceberAberto(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaReceberAberto(idclinica, idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaReceberBaixado(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaReceberBaixado(idclinica, idunidadeatendimento);
        }

        public decimal TotalContasaReceber(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaReceber(idclinica, idunidadeatendimento);
        }

        public decimal TotalContasaReceberHoje(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaReceberHoje(idclinica, idunidadeatendimento);
        }

        public decimal TotalContasaReceberAmanha(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaReceberAmanha(idclinica,idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaPagar(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaPagar(idclinica, idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaPagarAberto(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaPagarAberto(idclinica, idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaPagarBaixado(int idclinica,int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaPagarBaixado(idclinica,idunidadeatendimento);
        }

        public decimal TotalContasaPagar(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaPagar(idclinica, idunidadeatendimento);
        }

        public decimal TotalContasaPagarHoje(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaPagarHoje(idclinica, idunidadeatendimento);
        }

        public decimal TotalContasaPagarAmanha(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.TotalContasaPagarAmanha(idclinica, idunidadeatendimento);
        }
        
        public void SalvarCheque(Cheque cheque)
        {
            _financeiroRepository.SalvarCheque(cheque);
        }

        public void ExcluirCheque(Cheque cheque)
        {
            _financeiroRepository.ExcluirCheque(cheque);
        }

        public Cheque ObterChequePorId(int id)
        {
            return _financeiroRepository.ObterChequePorId(id);
        }

        public ICollection<Cheque> ListarCheques(int idClinica)
        {
            return _financeiroRepository.ListarCheques(idClinica);
        }

        public ICollection<Cheque> PesquisarCheques(string emitente, string banco, int idClinica)
        {
            return _financeiroRepository.PesquisarCheques(emitente, banco, idClinica);
        }

        public void SalvarBanco(Banco banco)
        {
            _financeiroRepository.SalvarBanco(banco);
        }

        public void ExcluirBanco(Banco banco)
        {
            _financeiroRepository.ExcluirBanco(banco);
        }

        public ICollection<Banco> ListarBancos()
        {
            return _financeiroRepository.ListarBancos();
        }

        public Banco ObterBancoPorId(int id)
        {
            return _financeiroRepository.ObterBancoPorId(id);
        }

        public void SalvarCentroCusto(CentroCusto centroCusto)
        {
            _financeiroRepository.SalvarCentroCusto(centroCusto);
        }

        public void ExcluirCentroCusto(CentroCusto centroCusto)
        {
            _financeiroRepository.ExcluirCentroCusto(centroCusto);
        }

        public CentroCusto ObterCentroCustoPorId(int id)
        {
            return _financeiroRepository.ObterCentroCustoPorId(id);
        }

        public ICollection<CentroCusto> ListarCentroCustos()
        {
            return _financeiroRepository.ListarCentroCustos();
        }
        
        public void TransferenciaConta(TransferenciaConta transferenciaConta)
        {
            _financeiroRepository.TransferenciaConta(transferenciaConta);
        }

        public void SalvarCaixaDiario(CaixaDiario caixaDiario)
        {
            throw new NotImplementedException();
        }

        public void ExcluirCaixaDiario(CaixaDiario caixaDiario)
        {
            _financeiroRepository.ExcluirCaixaDiario(caixaDiario);
        }

        public void SalvarMovimentacaoCaixa(MovimentoCaixa movimentoCaixa)
        {
            _financeiroRepository.SalvarMovimentacaoCaixa(movimentoCaixa);
        }

        public int QuantidadeRegistrosContasaReceber(int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.QuantidadeRegistrosContasaReceber(idclinica, idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaPagarPorId(int id, int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaPagarPorId(id,idclinica,idunidadeatendimento);
        }

        public ICollection<FinanceiroParcela> ListarContasaPagarPorCredor(string credor, int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarContasaPagarPorCredor(credor, idclinica, idunidadeatendimento);
        }

        public ICollection<MeioPagamento> ListarMeiosdePagamentos()
        {
            return _financeiroRepository.ListarMeiosdePagamentos();
        }

        public MeioPagamento ObterMeioPagamentoPorId(int id)
        {
            return _financeiroRepository.ObterMeioPagamentoPorId(id);
        }


        #region [Tranferencias entre contas]

        public List<TransferenciaConta> GetTransferencias()
        {
            return _financeiroRepository.GetTransferencias();
        }

        public TransferenciaConta GetTransferenciaById(int id)
        {
            return _financeiroRepository.GetTransferenciaById(id);
        }

        public void SalvarTransferencia(TransferenciaConta model)
        {
            _financeiroRepository.SalvarTransferencia(model);
        }

        public void ExcluirTransferencia(TransferenciaConta model)
        {
            _financeiroRepository.ExcluirTransferencia(model);
        }

        public FinanceiroParcela ObterFinanceiroParcelaPorId(int id)
        {
            return _financeiroRepository.ObterFinanceiroParcelaPorId(id);
        }

        public FinanceiroParcela AlterarParcela(FinanceiroParcela parcela)
        {
            return _financeiroRepository.AlterarParcela(parcela);
        }

        public ICollection<FinanceiroParcela> ListarParcelasPesquisa(string tipo, string descricao, string tipoConta, int idclinica, int idunidadeatendimento)
        {
            return _financeiroRepository.ListarParcelasPesquisa(tipo, descricao, tipoConta,idclinica,idunidadeatendimento);
        }

       
        #endregion
    }
}